### DISCUSSION TITLE
terraform and machine id

### DISCUSSION THREAD

**Discussion by redpimpernel** [gravitational/teleport#27830] [Q&A] [answered]
Hi all,
I have started trying out Teleport for my home lab environment. In hopes that I might use it for my small business eventually.
I have come to a sticking point and I do not understand the best way to continue.

I am trying to set up Teleport using Terraform and Ansible.
Terraform provisions a Linode VPS and then runs an Ansible playbook against the new VPS.
I would like Terraform to then configure Teleport with my user and roles. However, Terraform won't run without a user existing that it can log in as to access the Teleport API. I feel like machine id is the way around this, create a bot as the last task in the Ansible playbook and Terraform can use the bot user.

Is this at-least somewhat correct, or am I way off course here?

Thank you all for any help you can give.

**Comment by webvictim** [comment]
That's a reasonable idea, yes. Creating regular users through API is tricky with Teleport as you need to visit a link in a web browser to set a password and provide a second factor, whereas bot users aren't subject to these same limitations.

**Reply by redpimpernel** [reply]
Right, so I am on the right path. What my issue then is, how can I get Terraform to use the bot user?
I am having a bit of difficulty understand how to run the Teleport Terraform provider as the bot.
Any idea?

**Reply by webvictim** [reply, marked as answer]
Sorry for the delay in replying. 

I tested this out and wrote a guide here on how to do this: https://github.com/gravitational/teleport/discussions/28491
